Scripts/XRLineRendererBase.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using UnityEngine;
+
+/// <summary>
+/// A unified base class for the XR Line Renderer and XR Trail Renderer
+/// </summary>
+[RequireComponent(typeof(MeshRenderer))]
+[RequireComponent(typeof(MeshFilter))]
+[ExecuteInEditMode]
+public abstract class XRLineRendererBase : MonoBehaviour
+{
+    static readonly GradientColorKey k_DefaultStartColor = new GradientColorKey(Color.white, 0);
+    static readonly GradientColorKey k_DefaultEndColor = new GradientColorKey(Color.white, 1);
+    static readonly GradientAlphaKey k_DefaultStartAlpha = new GradientAlphaKey(1,0);
+    static readonly GradientAlphaKey k_DefaultEndAlpha = new GradientAlphaKey(1,1);
+
+    [SerializeField]
+    [Tooltip("Materials to use when rendering.")]
+    protected Material[] m_Materials;
+
+    [SerializeField]
+    [Tooltip("The multiplier applied to the curve, describing the width (in world space) along the line.")]
+    protected float m_Width = 1.0f;
+
+    [SerializeField]
+    [Tooltip("The curve describing the width of the line at various points along its length.")]
+    protected AnimationCurve m_WidthCurve = new AnimationCurve();
+
+    [SerializeField]
+    [Tooltip("The gradient describing color along the line.")]
+    protected Gradient m_Color = new Gradient();
+
+    [SerializeField]
+    [HideInInspector]
+    protected MeshRenderer m_MeshRenderer;
+
+    // Cached Data
+    protected XRMeshChain m_XRMeshData;
+    protected bool m_MeshNeedsRefreshing;
+    protected float m_StepSize = 1.0f;
+
+    /// <summary>
+    /// Set the width at the start of the line.
+    /// </summary>
+    public float widthStart
```

Comment:
What do we think about a .width as well that sets all keys? Same thing for .color, etc?